go/ethadapter/mgmtcontractlib/mgmt_contract_lib.go

```diff
@@ -119,20 +120,22 @@ func (c *contractLibImpl) DecodeTx(tx *types.Transaction) common.L1TenTransactio
 	return nil
 }
 
-// CreateBlobRollup creates a BlobTx, encoding the rollup data into blobs.
-func (c *contractLibImpl) CreateBlobRollup(t *common.L1RollupTx) (types.TxData, error) {
+// PopulateAddRollup creates a BlobTx, encoding the rollup data into blobs.
+func (c *contractLibImpl) PopulateAddRollup(t *common.L1RollupTx, blobs []*kzg4844.Blob) (types.TxData, error) {
 	decodedRollup, err := common.DecodeRollup(t.Rollup)
 	if err != nil {
 		panic(err)
 	}
 
 	metaRollup := ManagementContract.StructsMetaRollup{
-		Hash:               decodedRollup.Hash(),
+		Hash:               decodedRollup.Hash(), //TODO we don't actually use this anymore, do we still need it in here?
```

Comment:
@StefanIliev545 this could do with your eyes. since we sign over the composite hash, do we wanna use the `r.CompositeHash` in the `AddRollup` function on management contract instead?

Reply:
No. Management contract builds the hash itself from the data it is aware of and verifies against signature. If you provide hash and signature, then there is virtually no security. You can provide any hash with any data along with a valid previous signature for said hash and the check will pass.